bigbluebutton-html5/imports/api/audio/client/manager/index.js

```diff
@@ -1,11 +1,76 @@
 import BaseAudioBridge from '../bridge/base';
 import VertoBridge from '../bridge/verto';
 import SIPBridge from '../bridge/sip';
-
+
+class CallStates {
+  static get init() {
+    return "initialized state";
+  }
+  static get echo() {
+    return "do echo test state";
+  }
+  static get callIntoEcho() {
+    return "calling into echo test state";
+  }
+  static get inEchoTest() {
+    return "in echo test state";
+  }
+  static get joinVoiceConference() {
+    return "join voice conference state";
+  }
+  static get callIntoConference() {
+    return "calling into conference state";
+  }
+  static get inConference() {
+    return "in conference state";
+  }
+  static get transferToConference() {
+    return "joining from echo into conference state";
+  }
+  static get echoTestFailed() {
+    return "echo test failed state";
+  }
+  static get callToListenOnly() {
+    return "call to listen only state";
+  }
+  static get connectToListenOnly() {
+    return "connecting to listen only state";
+  }
+  static get inListenOnly() {
+    return "in listen only state";
+  }
+  static get reconnecting() {
+    return "reconecting";
+  }
+};
+
+const ErrorCodes = {
+  CODE_1001: "1001",
+  CODE_1002: "1002",
+  CODE_1003: "1003",
+  CODE_1004: "1004",
+  CODE_1005: "1005",
+  CODE_1006: "1006",
+  CODE_1007: "1007",
+  CODE_1008: "1008",
+  CODE_1009: "1009",
+  CODE_1010: "1010",
+  CODE_1011: "1011"
+};
+
+const AudioErrorCodes = Object.freeze(ErrorCodes);
+
 // manages audio calls and audio bridges
-export default class AudioManager {
-  constructor(userData) {
-    const MEDIA_CONFIG = Meteor.settings.public.media;
+class AudioManager {
+  constructor() {
+  }
+
+  init(userData) {
+    // this check ensures changing locales will not rerun init
+    if (this.currentState != undefined) {

```

Comment:
As is, null wouldnt work because its actually checking if the variable was defined. Not the value.

Because the class has no constructor It gets initialized with an init function. This had to be done to control the class because if its imported into the client, when locales cause the whole application to re render the class would get recreated again. 

There could always be a flag to check if the class has been created but not initialized yet. In this case checking that member works.

The class had to be created somewhere and only initialised once. But if you just import the class somewhere like main.jsx for example when the app reloads it would recreate it when you want that data in the class persisted